src/renderers/shared/fiber/ReactChildFiber.js

```diff
@@ -200,6 +200,16 @@ function throwOnInvalidObjectType(returnFiber: Fiber, newChild: Object) {
   }
 }
 
+function warnOnFunctionType() {
+  warning(
+    false,
```

Comment:
Why not pass in newChild and put `typeof newChild !== 'function'` instead of `false`?